cmd/kaspawallet/daemon/server/split_transaction.go

```diff
@@ -197,3 +222,37 @@ func (s *server) estimateMassAfterSignatures(transaction *serialization.Partiall
 
 	return s.txMassCalculator.CalculateTransactionMass(transactionWithSignatures), nil
 }
+
+func (s *server) moreUTXOsForMergeTransaction(alreadySelectedUTXOs []*libkaspawallet.UTXO, requiredAmount uint64) (
+	additionalUTXOs []*libkaspawallet.UTXO, totalValueAdded uint64, err error) {
+
+	dagInfo, err := s.rpcClient.GetBlockDAGInfo()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	for _, utxo := range s.utxosSortedByAmount {
+		if !isUTXOSpendable(utxo, dagInfo.VirtualDAAScore, s.params.BlockCoinbaseMaturity) {
+			continue
+		}
+		for _, alreadySelectedUTXO := range alreadySelectedUTXOs {
```

Comment:
Replying to
> I guess because that will require to implement a walletUTXO set, and that would be a premature optimization

You just need to convert it to a set of the type `map[externalapi.DomainOutpoint]struct{}`. No need to use `walletUTXOSet` or something like that.